project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/DataManagerVault.java

```diff
@@ -0,0 +1,198 @@
+package life.qbic.projectmanagement.infrastructure;
+
+import static life.qbic.logging.service.LoggerFactory.logger;
+
+import java.io.BufferedOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.KeyStore.PasswordProtection;
+import java.security.KeyStore.SecretKeyEntry;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.crypto.spec.SecretKeySpec;
+import life.qbic.logging.api.Logger;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Container for a Java keystore to maintain secrets within the application.
+ *
+ * @since 1.8.0
+ */
+@Component
+public class DataManagerVault implements DisposableBean {
+
+  public static final String UNEXPECTED_VAULT_EXCEPTION = "Unexpected vault exception";
+  private static final Logger log = logger(DataManagerVault.class);
+  private static final String KEY_GENERATOR_ALGORITHM = "AES";
+  private static final double MIN_ENTROPY = 100; // Shannon entropy * length of secret
+  private final KeyStore keyStore;
+  private final String envVarKeystorePassword;
+  private final String envVarKeystoreEntryPassword;
+  private final Path keystorePath;
+
+  public DataManagerVault(@Value("${qbic.security.vault.key.env}") String vaultKeyEnvVar,
+      @Value("${qbic.security.vault.path}") String vaultPathString,
+      @Value("${qbic.security.vault.entry.password.env}") String vaultEntryPassword)
+      throws CertificateException, KeyStoreException, IOException, NoSuchAlgorithmException {
+    if (System.getenv(vaultKeyEnvVar) == null) {
```

Comment:
Why do you use `System.getenv` here? Whether the value stems from an environment variable or other source is determined by the spring framework. You can also provide it in the application properties without environment variable.

You should check for null or isBlank here instead.